src/ddmd/expression.d

```diff
@@ -15682,6 +15682,12 @@ extern (C++) final class EqualExp : BinExp
         if (e1.op == TOKtype || e2.op == TOKtype)
             return incompatibleTypes();
 
+        auto t1 = e1.type;
```

Comment:
t1 and t2
are either already used or getting used futher down.
I recommend scopeing `{` them in a bock `}`